includes/core/ValueObjects/API/Request/class-create-charge.php

```diff
@@ -0,0 +1,111 @@
+<?php
+/**
+ * Create charge request.
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Core\ValueObjects\API\Request;
+
+use WCPay\Core\Enums\Http_Methods;
+use WCPay\Core\Enums\Wc_Pay_Endpoints;
+
+/**
+ * Create charge request.
+ */
+final class Create_Charge extends Base_Request {
+
+	/**
+	 * AMount to charge.
+	 *
+	 * @var int $amount
+	 */
+	protected $amount;
+
+	/**
+	 * ID of the source to associate with charge.
+	 *
+	 * @var string $source_id
+	 */
+	protected $source_id;
+
+	/**
+	 * Get amount.
+	 *
+	 * @return int
+	 */
+	public function get_amount() {
+		return $this->amount;
+	}
+
+	/**
+	 * Get source id
+	 *
+	 * @return string
+	 */
+	public function get_source_id() {
+		return $this->source_id;
+	}
+
+	/**
+	 * Set amount.
+	 *
+	 * @param int $amount Amount to charge.
+	 * @return self
+	 * @throws \InvalidArgumentException
+	 */
+	public function set_amount( $amount ) {
+		if ( ! is_numeric( $amount ) ) {
+			throw new \InvalidArgumentException( 'Amount is not numeric value' );
+		}
+		$this->amount = $amount;
+		return $this;
+	}
```

Comment:
I like the general approach that implementation classes can use setters to validate the fields.

`Base_Request` could provide a validator or validation methods to make things easier and more consistent. Something like 
```php
	public function set_amount( $amount ) {
		$this->validator->number($amount);
		$this->amount = $amount;
		return $this;
	}
```

Reply:
I wanted to do it like this, but was afraid that it will be too broad for this use case. One approach would be to use validator class somewhere from composer package or built our own validation methods. For now, I will add those into parent Request class, and maybe we could expand it with more validators.

Reply:
Done.